pySDC/projects/Resilience/collocation_adaptivity.py

```diff
@@ -0,0 +1,296 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.colors import TABLEAU_COLORS
+
+from pySDC.helpers.stats_helper import get_sorted
+from pySDC.projects.Resilience.vdp import run_vdp
+from pySDC.projects.Resilience.advection import run_advection
+from pySDC.projects.Resilience.heat import run_heat
+from pySDC.projects.Resilience.hook import LogData
+from pySDC.projects.Resilience.accuracy_check import get_accuracy_order
+from pySDC.implementations.convergence_controller_classes.adaptive_collocation import AdaptiveCollocation
+from pySDC.core.Hooks import hooks
+
+
+# define global parameters for running problems and plotting
+CMAP = list(TABLEAU_COLORS.values())
+
+
+Tend = 0.015
+base_params = {
+    'step_params': {'maxiter': 99},
+    'sweeper_params': {
+        'QI': 'LU',
+        'num_nodes': 4,
+    },
+    'level_params': {'restol': 1e-9, 'dt': Tend},
+}
+
+coll_params_inexact = {
+    'num_nodes': [2, 3, 4],
+    'restol': [1e-4, 1e-7, 1e-9],
+}
+coll_params_refinement = {
+    'num_nodes': [1, 2, 3, 4],
+}
+coll_params_reduce = {
+    'num_nodes': [4, 3, 2, 1],
+}
+coll_params_type = {
+    # 'quad_type': ['RADAU-RIGHT', 'GAUSS'],
+    'quad_type': ['GAUSS', 'RADAU-RIGHT', 'LOBATTO'],
+}
+
+special_params = {
+    'inexact': {AdaptiveCollocation: coll_params_inexact},
+    'refinement': {AdaptiveCollocation: coll_params_refinement},
+    'reduce': {AdaptiveCollocation: coll_params_reduce},
+    'standard': {},
+    'type': {AdaptiveCollocation: coll_params_type},
+}
+
+
+def get_collocation_order(quad_type, num_nodes, node_type):
+    """
+    Compute the maximal order achievable by a given collocation method
+    """
+    pass
+
+
+# define a few hooks
+class LogSweeperParams(hooks):
+    """
+    Log the sweeper parameters after every iteration to check if the collocation inexactness is doing what it's supposed
+    to.
+    """
+
+    def post_iteration(self, step, level_number):
+        """
+        Args:
+            step (pySDC.Step.step): the current step
+            level_number (int): the current level number
+
+        Returns:
+            None
+        """
+        super().post_iteration(step, level_number)
+
+        L = step.levels[level_number]
+
+        self.add_to_stats(
+            process=step.status.slot,
+            time=L.time,
+            level=L.level_index,
+            iter=step.status.iter,
+            sweep=L.status.sweep,
+            type='sweeper_params',
+            value=L.sweep.params.__dict__,
+        )
+        self.add_to_stats(
+            process=step.status.slot,
+            time=L.time,
+            level=L.level_index,
+            iter=step.status.iter,
+            sweep=L.status.sweep,
+            type='coll_order',
+            value=L.sweep.coll.order,
+        )
+
+
+class LogLocalErrorPostIter(hooks):
+    """
```

Comment:
Sounds like a hook class which could be used for more examples? Make this generic?